pydantic_ai_slim/pydantic_ai/providers/xai.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import annotations as _annotations
+
+import os
+from typing import overload
+
+from pydantic_ai import ModelProfile
+from pydantic_ai.exceptions import UserError
+from pydantic_ai.profiles.grok import grok_model_profile
+from pydantic_ai.providers import Provider
+
+try:
+    from xai_sdk import AsyncClient
+    from xai_sdk.types.model import AllModels
+except ImportError as _import_error:  # pragma: no cover
+    raise ImportError(
+        'Please install the `xai-sdk` package to use the xAI provider, '
+        'you can use the `xai` optional group — `pip install "pydantic-ai-slim[xai]"`'
+    ) from _import_error
+
+
+# https://docs.x.ai/docs/models
+XaiModelName = str | AllModels
```

Comment:
With other providers, we have this in the model file instead, so let's move it just for consistency, unless there's a particular reason we cant